src/redux/actions/pipeline/runGem2s.js

```diff
@@ -7,24 +8,26 @@ import {
 } from '../../actionTypes/experimentSettings';
 
 import {
-  BACKEND_STATUS_LOADING,
-  BACKEND_STATUS_ERROR,
+  BACKEND_STATUS_ERROR, BACKEND_STATUS_LOADED,
 } from '../../actionTypes/backendStatus';
 
-import loadBackendStatus from '../backendStatus/loadBackendStatus';
-
 const runGem2s = (experimentId) => async (dispatch, getState) => {
-  const { experiments } = getState();
-
-  const projectId = experiments[experimentId].projectUuid;
+  const { experiments, backendStatus } = getState();
 
+  const runningStatus = {
+    ...backendStatus[experimentId].status,
+    gem2s: {
+      status: pipelineStatus.RUNNING,
+      completedSteps: [],
+    },
+  };
   dispatch({
-    type: BACKEND_STATUS_LOADING,
+    type: BACKEND_STATUS_LOADED,
```

Comment:
Similar thing here, we could have the backend status reducer react to EXPERIMENT_SETTINGS_PIPELINE_START (or EXPERIMENT_SETTINGS_GEM2S_START as a new action, I am not sure what is the best option here) and as a consequence set itself to the status you want in the reducer.

Moving this logic to a reducer will also help with this https://redux.js.org/style-guide/style-guide#put-as-much-logic-as-possible-in-reducers